test/suites/secondary_http.bash

```diff
@@ -0,0 +1,87 @@
+start_http_server() {
+    local host="127.0.0.1"
+    local port="8080"
+    local url="http://${host}:${port}/"
+    export CCACHE_SECONDARY_STORAGE="${url}"
+
+    mkdir "secondary"
+    echo "=== TEST ${CURRENT_TEST} ===" >>http-server.log
+    "${HTTP_SERVER}" --bind "${host}" --directory "secondary" "${port}" >>http-server.log 2>&1 &
+
+    echo "=== TEST ${CURRENT_TEST} ===" >>http-client.log
+    "${HTTP_CLIENT}" "${url}" >>http-client.log 2>&1 || test_failed_internal "Cannot connect to server"
+}
+
+SUITE_secondary_http_PROBE() {
+    if ! "${HTTP_SERVER}" --help >/dev/null 2>&1; then
+        echo "cannot execute ${HTTP_SERVER} - Python 3 might be missing"
+    fi
+}
+
+SUITE_secondary_http_SETUP() {
+    unset CCACHE_NODIRECT
+
+    start_http_server
+
+    generate_code 1 test.c
+}
+
+SUITE_secondary_http_TEARDOWN() {
+    terminate_all_children
```

Comment:
Would it make sense to let `test/run` always call `terminate_all_children` after a test suite has been run?

Reply:
Added it behind the `_TEARDOWN` invocation.